membership-attribute-service/app/controllers/AccountController.scala

```diff
@@ -40,12 +41,77 @@ class AccountController extends Controller with LazyLogging {
   lazy val mmaAction = NoCacheAction andThen mmaCorsFilter andThen BackendFromCookieAction
   lazy val mmaCardAction = NoCacheAction andThen corsCardFilter andThen BackendFromCookieAction
 
+
+
+  def cancelSubscription [P <: SubscriptionPlan.AnyPlan : SubPlanReads] = mmaCardAction.async { implicit request =>
+
+    val tp = request.touchpoint
+    val cancelForm = Form { single("reason" -> nonEmptyText) }
+    val maybeUserId = authenticationService.userId
+
+    def handleInputBody(cancelForm: Form[String]): Future[ApiError \/ String] = Future.successful {
+      cancelForm.bindFromRequest().value.map { cancellationReason =>
```

Comment:
shouldn't this be a json post instead of a regular form ? how is this going to be called ?

Reply:
Good call, I will change the format of the post to be JSON only (if it is possible) in a future PR.